syntax = "proto3";

package langchain;
option go_package = "./gen";

service LangChain {
  rpc Summarize(SummarizeRequest) returns (SummarizeResponse);
  rpc CleanWebpage(CleanWebpageRequest) returns (CleanWebpageResponse);
  rpc Chat(ChatRequest) returns (ChatResponse);
}

message ChatRequest {
  string message = 1;
}

message ChatResponse {
  string response = 1;
}

message CleanWebpageRequest {
  string content = 1;
  string description = 2;
}

message CleanWebpageResponse {
  string content = 1;
}

message SummarizeRequest {
  string content = 1;
  string query = 2;
}

message SummarizeResponse {
  string summary = 1;
}
